app/src/main/java/pm/gnosis/heimdall/ui/addressbook/add/AddressBookAddEntryActivity.kt

```diff
@@ -29,80 +25,65 @@ import pm.gnosis.utils.exceptions.InvalidAddressException
-        inject()
-        setContentView(R.layout.layout_address_book_add_entry)
-        setupToolbar(layout_add_address_book_entry_toolbar)
-    }
+    override fun screenId() = ScreenId.ADDRESS_BOOK_ENTRY
 
     override fun onStart() {
         super.onStart()
+        disposables += toolbarHelper.setupShadow(layout_address_book_update_entry_toolbar_shadow, layout_address_book_update_entry_scroll_view)
 
-        disposables += layout_add_address_book_entry_scan.clicks()
+        disposables += layout_address_book_update_entry_scan.clicks()
             .subscribeBy(onNext = { QRCodeScanActivity.startForResult(this) })
 
-        disposables += layout_add_address_book_entry_address.textChanges()
-            .subscribeBy(onNext = { layout_add_address_book_entry_address_container.error = null })
-
-        disposables += layout_add_address_book_entry_name.textChanges()
-            .subscribeBy(onNext = { layout_add_address_book_entry_name_container.error = null })
-
-        disposables += layout_add_address_book_entry_save.clicks()
+        disposables += layout_address_book_update_entry_save.clicks()
             .flatMap {
                 viewModel.addAddressBookEntry(
-                    layout_add_address_book_entry_address.text.toString(),
-                    layout_add_address_book_entry_name.text.toString(),
-                    layout_add_address_book_entry_description.text.toString()
+                    layout_address_book_update_entry_address.text.toString(),
+                    layout_address_book_update_entry_name.text.toString()
                 )
             }
             .observeOn(AndroidSchedulers.mainThread())
             .subscribeForResult(onNext = ::onAddressBookEntryAdded, onError = ::onAddressBookEntryAddError)
+
+        disposables += layout_address_book_update_entry_back_arrow.clicks()
+            .subscribeBy(onNext = { onBackPressed() }, onError = Timber::e)
     }
 
     private fun onAddressBookEntryAdded(entry: AddressBookEntry) {
-        toast("Added ${entry.name}")
         finish()
     }
 
     private fun onAddressBookEntryAddError(throwable: Throwable) {
         Timber.e(throwable)
-        when (throwable) {
-            is InvalidAddressException -> layout_add_address_book_entry_address_container.error = getString(R.string.invalid_ethereum_address)
-            is AddressBookContract.NameIsBlankException -> layout_add_address_book_entry_name_container.error =
-                    getString(R.string.name_cannot_be_blank)
-            is AddressBookContract.AddressAlreadyAddedException -> layout_add_address_book_entry_address_container.error =
-                    getString(R.string.address_already_in_address_book)
-            else -> snackbar(layout_add_address_book_entry_coordinator, getString(R.string.unknown_error))
-        }
+        toast(
+            when (throwable) {
+                is InvalidAddressException -> R.string.invalid_ethereum_address
```

Comment:
Why don'T we map it in the viewmodel to a SimpleLocalizedException and use `errorSnackbar()`?